packages/@aws-cdk/pipelines/lib/stage.ts

```diff
@@ -243,24 +243,32 @@ export class CdkStage extends CoreConstruct {
 
       for (const entry of manifest.entries) {
         let assetType: AssetType;
+        let assetPublishingRoleArn: string | undefined;
```

Comment:
both branches of the conditional produce the same value, why not just set it here?

Reply:
Because the `destination` property only exists on the specific subclasses, not the top-level interface (`IManifestEntry`). I was avoiding changing that, as the design there looks very intentional; however, I'm noticing now this is all part of `pipelines/private`, so I should be able to muck with it without any public-facing impacts.